### PR TITLE
Feature/tarjan

### PR DESCRIPTION
**Pull request by monkin77**

### CODE REVIEW

File: src/bakery/Interface.cpp [hunk 1]
```diff
@@ -53,10 +53,11 @@ void Interface::servicePlanner() {
     cout << "[1]: Single van visiting all clients without delivery times or capacity" << endl;
     cout << "[2]: Single van visiting all clients with delivery times but no capacity" << endl;
     cout << "[3]: Multiple vans with client allocation, capacity and delivery times" << endl;
+    cout << "[4]: SCC component of the graph containing the Bakery" << endl;
```
Comment (BrunoRosendo) [inline src/bakery/Interface.cpp]: It's fine if this option exists but we still need to adapt the SCC algorithm used to the graph (directed/undirected) and apply the right one in the algorithms. We should ask the user if the graph is directed or not, and maybe put a flag in the input file too

File: src/bakery/Bakery.cpp [hunk 2]
```diff
@@ -320,3 +320,11 @@ void Bakery::solveThirdPhase(bool useKnapsack, bool optimize) {
...
     return vans;
 }
+
+void Bakery::calculateSCC() {
+    this->graph.calculateSccTarjan(this->startingVertex);
+}
+
+void Bakery::displayAllScc() {
+    this->graph.displaySccTarjan();
+}
```
Comment (BrunoRosendo) [inline src/bakery/Bakery.cpp]: These functions seem unnecessary. I would just call the SCC function from graph and apply the right algorithms there

File: src/bakery/Interface.cpp (hunk 1 above)
Comment (BrunoRosendo) [reply src/bakery/Interface.cpp]: Also, we should probably show it with GraphViewer instead (another PR I guess)

File: src/graph/Graph.cpp [hunk 3]
```diff
@@ -351,3 +351,97 @@ void Graph::addPathToEdgeList(vector<Edge> &edges, Vertex* source, Vertex* dest)
...
+    int w = 0;  // To store stack extracted vertices
+    if(low[u] == disc[u]) {     // Head node of the SCC
+        while(st.top() != u) {  // Traverse the stack from the last element until the first of the SCC
+            w = st.top();
+            Vertex* currVertex = this->findVertex(w);   // possibly store currVertex
+            cout << w << " ";
+            stackMember[w] = false;
+            st.pop();
+        }
+        cout << endl << endl;
```
Comment (BrunoRosendo) [inline src/graph/Graph.cpp]: Is it intended to print stuff even when we're just calculation the SCC?

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]
The algorithm itself seems to be working fine but I have some suggestions. Also, the print solution function is still getting called with option 4:
image

File: src/graph/Graph.cpp (hunk 3 above)
Comment (monkin77) [reply src/graph/Graph.cpp]: I'm still making some changes, but I plan to remove it

**Comment by monkin77** [comment]
The algorithm seems to be working now. However, when using real maps (Porto and such) it throws errors so we still need to see what's wrong with that

**Review by BrunoRosendo — APPROVED** [review]
Looks fine. Keep up the good work

**Pull request merged**